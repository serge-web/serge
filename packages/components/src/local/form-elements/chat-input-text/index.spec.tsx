/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import ChatInputText from './index'

const createNodeMock = (el: any): HTMLTextAreaElement | null => {
  if (el.type === 'textarea') {
    return document.createElement('textarea')
  }
  return null
}

describe('ChatInputText component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ChatInputText placeholder={'Input text here'} />,
        { createNodeMock }
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
