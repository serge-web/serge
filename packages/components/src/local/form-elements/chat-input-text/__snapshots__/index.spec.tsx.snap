// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInputText component: renders correctly 1`] = `
<div
  className="main"
>
  <section
    className="input-container disable-offset fullwidth"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      onKeyDown={[Function]}
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root makeStyles-root-1 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline makeStyles-multiline-3 MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
        onClick={[Function]}
      >
        <textarea
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input makeStyles-input-2 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
          disabled={false}
          name=""
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Input text here"
          required={false}
          rows={2}
          style={
            Object {
              "height": 0,
              "overflow": "hidden",
            }
          }
        />
        <textarea
          aria-hidden={true}
          className="MuiInputBase-input MuiFilledInput-input makeStyles-input-2 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
          readOnly={true}
          style={
            Object {
              "height": 0,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "transform": "translateZ(0)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        />
        <div
          className="MuiInputAdornment-root action MuiInputAdornment-positionEnd MuiInputAdornment-filled"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root icon-send MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4.01 6.03l7.51 3.22-7.52-1 .01-2.22m7.5 8.72L4 17.97v-2.22l7.51-1M2.01 3L2 10l15 2-15 2 .01 7L23 12 2.01 3z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <hr
            className="MuiDivider-root divider"
          />
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root icon-cancel"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
